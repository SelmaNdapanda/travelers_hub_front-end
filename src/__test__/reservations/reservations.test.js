import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import Reservations from '../../components/reservations/reservations';
import '@testing-library/jest-dom';

describe('Reservation component renders', () => {
  it('renders the reservations', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Reservations />
        </BrowserRouter>
      </Provider>,
    );

    expect(
      screen.getByText(/There are no bookings to display/),
    ).toBeInTheDocument();
  });

  it('It matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Reservations />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
