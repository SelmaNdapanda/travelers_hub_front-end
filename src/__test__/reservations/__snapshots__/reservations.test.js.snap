// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation component renders It matches the snapshot 1`] = `
<main
  className="reservation-main"
>
  <div
    className="filter-options"
  >
    <select
      id="filter-by"
      onChange={[Function]}
    >
      <option>
        All
      </option>
      <option>
        My
      </option>
    </select>
    <input
      data-testid="search-bar"
      id="search-bar"
      onChange={[Function]}
      placeholder="Search reservations..."
    />
  </div>
  <section
    className="reservations-holder"
  >
    <p>
      There are no bookings to display
    </p>
  </section>
</main>
`;
